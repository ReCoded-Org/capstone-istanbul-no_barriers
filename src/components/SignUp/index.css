:root {
  --title-color: #002736;
  --subTitle-color: #000000;
  --background-color: #fc9e01;
  --text-color: #eff4f6;
  --filter-shadow-color: #b3b3b3;
  --link-color: #0641af;
}

.webLogo,
.welcometext {
  color: var(--title-color);
  margin-top: 2rem;
}

.signupBtn {
  background-color: var(--background-color);
  font-weight: 600;
  cursor: pointer;
  width: 7rem;
  height: 3rem;
  font-size: 1rem;
  border-radius: 2.1875rem;
  border-style: hidden;
  box-shadow: 0rem 0.6875rem 1.5rem -0.4375rem var(--filter-shadow-color);
}

.loginLink {
  display: inline;
  color: var(--link-color);
}
