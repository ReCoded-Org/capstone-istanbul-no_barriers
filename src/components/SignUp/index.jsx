import React, { useState } from "react";
import { Row, Form, Col, Button, Container, Alert } from "react-bootstrap";
import { Redirect } from "react-router-dom";
import { auth } from "../../firebaseConfig";
import "./index.css";
import { useTranslation } from "react-i18next";

const SignUp = () => {
  const { t } = useTranslation();
  const [redirect, setRedirect] = useState(false);
  const [alert, setAlert] = useState(null);
  const [signupInfo, setSignupInfo] = useState({
    firstName: "",
    lastName: "",
    email: "",
    password: "",
    repeatPassword: "",
  });

  const handleSignup = async (e) => {
    e.preventDefault();
    if (signupInfo.password !== signupInfo.repeatPassword) {
      setAlert({ message: t("signup.alertMsg"), status: "danger" });
      return;
    }
    try {
      const response = await auth.createUserWithEmailAndPassword(
        signupInfo.email,
        signupInfo.password
      );
      response.user.sendEmailVerification();
      response.user.updateProfile({
        displayName: signupInfo.firstName + " " + signupInfo.lastName,
      });
    } catch (error) {
      setAlert({ message: error.message, status: "danger" });
      return;
    }
    setSignupInfo({
      firstName: "",
      lastName: "",
      email: "",
      password: "",
      repeatPassword: "",
    });
    setAlert({ message: t("signup.alertCmplt"), status: "success" });
    setTimeout(() => setRedirect(true), 1000);
  };

  const handleChange = (key) => (e) => {
    setSignupInfo({ ...signupInfo, [key]: e.target.value });
  };

  return (
    <Container className="signupContainer">
      <Row className="webLogo justify-content-center">
        <h2>No Barriers</h2>
      </Row>
      <Row className="justify-content-center">
        <h4 className="welcometext">{t("signup.welcomeMsg")}</h4>
      </Row>
      <Row className="justify-content-center">
        <Form onSubmit={handleSignup} className="form">
          <Form.Row>
            <Form.Group as={Col} controlId="formGridFirstName" className="mt-3">
              <Form.Control
                required
                type="name"
                placeholder={t("signup.firstName")}
                value={signupInfo.firstName}
                onChange={handleChange("firstName")}
              />
            </Form.Group>

            <Form.Group as={Col} controlId="formGridLastName" className="mt-3">
              <Form.Control
                required
                type="name"
                placeholder={t("signup.lastName")}
                value={signupInfo.lastName}
                onChange={handleChange("lastName")}
              />
            </Form.Group>
          </Form.Row>

          <Form.Group controlId="formGridEmail" className="mt-3">
            <Form.Control
              type="email"
              placeholder={t("signup.email")}
              value={signupInfo.email}
              onChange={handleChange("email")}
            />
          </Form.Group>

          <Form.Row>
            <Form.Group as={Col} controlId="formGridPassword" className="mt-3">
              <Form.Control
                required
                type="password"
                placeholder={t("signup.password")}
                value={signupInfo.password}
                onChange={handleChange("password")}
              />
            </Form.Group>

            <Form.Group
              as={Col}
              controlId="formGridRepeatPassword"
              className="mt-3"
            >
              <Form.Control
                required
                type="password"
                placeholder={t("signup.repeatPassword")}
                value={signupInfo.repeatPassword}
                onChange={handleChange("repeatPassword")}
              />
            </Form.Group>
          </Form.Row>
          <Form.Row className="justify-content-center">
            <Button variant="primary" type="submit" className="signupBtn mt-3">
              {t("signup.signup")}
            </Button>
          </Form.Row>
        </Form>
      </Row>
      <Row className="justify-content-center"></Row>
      {alert && (
        <Alert
          className="mt-3"
          variant={alert.status}
          onClose={() => setAlert(null)}
          dismissible
        >
          <p>{alert.message}</p>
        </Alert>
      )}
      {redirect && <Redirect to="/" />}
      <Row className="justify-content-center">
        <p className="mt-3">
          {t("signup.existingAccount")}
          <a href="/login" className="loginLink">
            {t("signup.login")}
          </a>
        </p>
      </Row>
    </Container>
  );
};

export default SignUp;
