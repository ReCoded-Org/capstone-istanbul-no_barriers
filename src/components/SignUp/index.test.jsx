import React from "react";
import renderer from "react-test-renderer";
import SignUp from "./index";

it("renders correctly", () => {
  const tree = renderer.create(<SignUp />).toJSON();
  expect(tree).toMatchSnapshot();
});
