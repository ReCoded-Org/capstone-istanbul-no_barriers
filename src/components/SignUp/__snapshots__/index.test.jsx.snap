// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="signupContainer container"
>
  <div
    className="webLogo justify-content-center row"
  >
    <h2>
      No Barriers
    </h2>
  </div>
  <div
    className="justify-content-center row"
  >
    <h4
      className="welcometext"
    >
      signup.welcomeMsg
    </h4>
  </div>
  <div
    className="justify-content-center row"
  >
    <form
      className="form"
      onSubmit={[Function]}
    >
      <div
        className="form-row"
      >
        <div
          className="mt-3 form-group col"
        >
          <input
            className="form-control"
            id="formGridFirstName"
            onChange={[Function]}
            placeholder="signup.firstName"
            required={true}
            type="name"
            value=""
          />
        </div>
        <div
          className="mt-3 form-group col"
        >
          <input
            className="form-control"
            id="formGridLastName"
            onChange={[Function]}
            placeholder="signup.lastName"
            required={true}
            type="name"
            value=""
          />
        </div>
      </div>
      <div
        className="mt-3 form-group"
      >
        <input
          className="form-control"
          id="formGridEmail"
          onChange={[Function]}
          placeholder="signup.email"
          type="email"
          value=""
        />
      </div>
      <div
        className="form-row"
      >
        <div
          className="mt-3 form-group col"
        >
          <input
            className="form-control"
            id="formGridPassword"
            onChange={[Function]}
            placeholder="signup.password"
            required={true}
            type="password"
            value=""
          />
        </div>
        <div
          className="mt-3 form-group col"
        >
          <input
            className="form-control"
            id="formGridRepeatPassword"
            onChange={[Function]}
            placeholder="signup.repeatPassword"
            required={true}
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        className="justify-content-center form-row"
      >
        <button
          className="signupBtn mt-3 btn btn-primary"
          disabled={false}
          type="submit"
        >
          signup.signup
        </button>
      </div>
    </form>
  </div>
  <div
    className="justify-content-center row"
  />
  <div
    className="justify-content-center row"
  >
    <p
      className="mt-3"
    >
      signup.existingAccount
      <a
        className="loginLink"
        href="/login"
      >
        signup.login
      </a>
    </p>
  </div>
</div>
`;
